File: CongNgheNET/TH/02.Bai Thuc Hanh ASPNET-MVC3 Helpers Forms Va Validation/Begin/VB/MvcMusicStore/Controllers/StoreController.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Public Class StoreController
    Inherits Controller
    Private storeDB As New MusicStoreEntities
    '
    ' GET: /Store/

    Public Function Index() As ActionResult
        'Create list of genres
        Dim genres = From genre In storeDB.Genres
                     Select genre.Name

        'Create your view model
        Dim viewModel = New StoreIndexViewModel With
                        {.Genres = genres.ToList(),
                         .NumberOfGenres = genres.Count()}

        Return View(viewModel)
    End Function

    '
    'GET: /Store/Browse?genre=Disco

    Public Function Browse(ByVal genre As String) As ActionResult

        Dim genreModel = storeDB.Genres.Include("Albums").Single(
            Function(g) g.Name = genre)

        Dim viewModel = New StoreBrowseViewModel With
                        {.Genre = genreModel,
                         .Albums = genreModel.Albums.ToList()}

        Return View(viewModel)
    End Function

    '
    'GET: /Store/Details/5

    Public Function Details(ByVal id As Integer) As ActionResult

        Dim album = storeDB.Albums.Single(Function(a) a.AlbumId = id)

        Return View(album)
    End Function
End Class

File: CongNgheNET/TH/[12520026][TH1][MvcMusicStore]/MvcMusicStore/Controllers/StoreController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using MvcMusicStore.ViewModels;
using MvcMusicStore.Models;

namespace MvcMusicStore.Controllers
{
    public class StoreController : Controller
    {
        public ActionResult Index()
        {
            // Create a list of genres
            var genres = new List<string> { "Rock", "Jazz", "Country", "Pop", "Disco" };

            // Create our view model
            var viewModel = new StoreIndexViewModel
            {
                NumberOfGenres = genres.Count(),
                Genres = genres
            };

            ViewBag.Starred = new List<string> { "Rock", "Jazz" };

            return View(viewModel);
        }

        public ActionResult Browse(string genre)
        {
            var genreModel = new Genre()
            {
                Name = genre
            };

            var albums = new List<Album>()
            {
                new Album() { Title = genre + " Album 1" },
                new Album() { Title = genre + " Album 2" }
            };

            var viewModel = new StoreBrowseViewModel()
            {
                Genre = genreModel,
                Albums = albums
            };

            return View(viewModel);
        }

        public ActionResult Details(int id)
        {
            var album = new Album { Title = "Sample Album" };

            return View(album);
        }

        //
        // GET: /Store/

        //public string Index()
        //{
        //    return "Hello from Store.Index()";
        //}

        //
        // GET: /Store/Browse

        //public string Browse()
        //{
        //    return "Hello from Store.Browse()";
        //}

        //
        // GET: /Store/Details

        //public string Details()
        //{
        //    return "Hello from Store.Details()";
        //}

        //
        // GET: /Store/Browse?genre=Disco

        //public string Browse(string genre)
        //{
        //    string message = HttpUtility.HtmlEncode("Store.Browse, Genre = " + genre);

        //    return message;
        //}

        //public string Details(int id)
        //{
        //    string message = "Store.Details, ID = " + id;

        //    return message;
        //}
    }
}
